crates/types/src/storage.rs

```diff
@@ -99,7 +99,7 @@ impl StorageCodec {
 ///
 /// # Important
 /// The [`Self::encode`] implementation should use the codec specified by [`Self::DEFAULT_CODEC`].
-pub trait StorageEncode {
+pub trait StorageEncode: Sized {
```

Comment:
Is this a preparation for introducing the `serialized_size` method?

Reply:
I just need it transitively. This gets removed in a follow up PR where I actually make this trait object-safe.